### PR TITLE
Add extra getters played, buffered, seekable

### LINKED ISSUE #61: Disable forward seeking
## Expected Behavior

It actually works as documented.  So this is just a question I guess.  I would like to be able to find the furthest time played before seeking.  The reason is so that I can insure users may not seek forward, only backward.
### Actual Behavior

When seeked happens, timeupdate also occurs (per the documentation).  But this is causing me trouble figuring out the time right before the seek happens so I can be sure not to seek past this point.
### Steps to Reproduce

I have come up with a solution, but it seems like there must be a better way.

```
var curtime = 0;

player.on('timeupdate', function(data) {
    if(data.seconds < curtime + 1 && data.seconds > curtime) {
    // Above is where I hack it.  I only update the current time if the timeupdate is less than a second ago (therefore proabaly not seeked to).  This stops the function from just updating curtime to the seeked time.  But I feel it's not the best way.
        curtime = data.seconds;
    }
});

player.on('seeked', function(data) {
    if(data.seconds > curtime) {
        player.setCurrentTime(curtime);
    }
});
```

### PR DESCRIPTION
**Pull request by luwes**
This PR adds new getters that will return the `video.played`, `video.buffered` and `video.seekable` read-only properties.

These are `TimeRanges` objects originally but since they hold functions it's not possible to pass those objects via `postMessage`.

To get around this they are serialized as arrays of [start, end].

The `seeking` event is already added FYI.

Fixes #61

### CODE REVIEW

**Comment by fisherinnovation** [comment]
@luwes Can you add JSDoc's to these new methods?

**Comment by fisherinnovation** [comment]
Can you also append these methods to the `player object includes all api methods` test. It will ensure we stay above the code coverage target.

**Comment by jybleau** [comment]
@luwes , I also mentioned about the `seeking` "property", not to be confused with the `seeking` event...
Is it possible to add it too? Thanks.

**Comment by luwes** [comment]
@jybleau that will need another PR I think, it's not added to the player core yet.

File: README.md [hunk 1]
```diff
@@ -855,6 +871,30 @@ player.setPlaybackRate(0.5).then(function(playbackRate) {
 });
 ```
 
+### getPlayed(): Promise&lt;array, Error&gt;
+
+Get the played ranges of the video.
+
+```js
+player.getPlayed().then(function(played) {
+    // played = array values of the played ranges.
```
Comment (tenacex) [inline README.md]: This comment is unclear

**Pull request merged**